js/main.js

```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
```

Comment:
you can talk to [David Walsh](https://davidwalsh.name/javascript-debounce-function) :yum: